lambda/translcore.ml

```diff
@@ -152,18 +152,23 @@ let rec push_defaults loc bindings cases partial =
   | _ ->
       cases
 
-let comprehension_prim ~is_concat ~dir= 
+let from_to_comp_prim ~is_concat ~dir ~single_loop= 
   let prefix = if is_concat then "concat_" else "" in
-  let suffix = match dir with 
+  let suffix = if single_loop then "" else "_prepend" in
+  let function_name = match dir with 
     | Upto ->  "map_from_to"
     | Downto -> "map_from_downto"  in 
-  let fun_name =  prefix ^ suffix in
+  let fun_name =  String.concat "" [prefix;function_name;suffix] in
   Lambda.transl_prim "CamlinternalComprehension" fun_name
 
-let list_comp_in_prim ~is_concat = 
+let in_from_to_comp_prim ~is_concat ~single_loop = 
```

Comment:
`in` and `from_to` is a bit confusing. Isn't it only used for the `in` cases?

Reply:
Yes.